src/test/unit/standalone/string.spec.ts

```diff
@@ -173,6 +174,25 @@ describe("basic string parsers", () => {
         });
     });
 
+    describe("caseString(foo)", () => {
+        const s = "foo";
+        const parser = caseString(s);
```

Comment:
suggestion: let's change this to have an explicit type annotation. This way it can also act as a test for the expected type.

The expected type is currently correct and defaults to the narrow constant type `"foo"` instead of `string`. However, if someone accidentally breaks that type narrowing feature in the future, an explicit annotation will catch that regression and also catch that bug

```suggestion
        const parser: Parjser<"foo"> = caseString("foo");
```